arangod/Cluster/CreateCollection.cpp

```diff
@@ -194,3 +194,13 @@ bool CreateCollection::first() {
 
   return false;
 }
+
+
+void CreateCollection::setState(ActionState state) {
+  
+  TRI_ASSERT(_description.has("shard"));
+  _feature.incShardVersion(_description.get("shard"));
```

Comment:
setState() gets called more than once.  incShardVersion() needs to happen only upon reaching a terminal state:  

if ((COMPLETE==state || FAILED==state) && _state != state)

Reply:
of course. crap.

Reply:
Actually, this code should add an attribute to the description such that it changes the hash.  Again, only when object is moving to terminal state.